energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -130,6 +196,67 @@ where
         ManagedVec::from_single_item(EsdtTokenPayment::new(reward_token_id, 0, total_rewards))
     }
 
+    // User rewards formula = user_base_rewards +
+    // min(base_const * user_base_rewards, boosted_rewards * (energy_const * user_energy / total_energy + farm_const * user_farm / total_farm) / (X+Y))
+    fn get_user_rewards_for_week(
+        &self,
+        module: &Self::WeeklyRewardsSplittingMod,
+        energy_amount: &BigUint<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+        total_energy: &BigUint<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+        total_rewards: &PaymentsVec<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+    ) -> PaymentsVec<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api> {
+        let mut user_rewards = ManagedVec::new();
+        let factors = module.boosted_yields_factors().get();
+        if energy_amount == &0
+            || energy_amount < &factors.min_energy_amount
+            || self.user_farm_amount < factors.min_farm_amount
+        {
+            return user_rewards;
+        }
+
+        for weekly_reward in total_rewards {
+            // normalized user rewards
+            let user_base_rewards_per_block =
+                &self.total_rewards_per_block * &self.user_farm_amount / &self.total_farm_supply;
+            let normalized_user_base_rewards =
```

Comment:
move these 2 before the for cycle.
rename normalized_user_base_rewards to user_rewards_for_week